target/linux/armvirt/patches-5.10/220-reverse-arm-gc_sections-armvirt-only.patch

```diff
@@ -0,0 +1,138 @@
+From ecf43bcf36083b5b7e02d87e1ee1beafa92d152d Mon Sep 17 00:00:00 2001
+From: Mathew McBride <matt@traverse.com.au>
+Date: Mon, 31 Jan 2022 03:55:04 +0000
+Subject: [PATCH] Revert "use -ffunction-sections, -fdata-sections and
+ --gc-sections"
+
+This patch conflicts with the generation of the EFI STUB for 32-bit
+ARM.
```

Comment:
Probably only one or some of these strips are breaking the arm32 bit kernel. Do you know which ones?

Reply:
It's a well above my knowledge level, but I think one or both of gc-sections and fdata-sections.
It might be that the vmlinux.ld.S for arm32 needs adjusting, as has been suggested for other platforms (arm64 https://groups.google.com/g/clang-built-linux/c/uamqmR5bHOc/m/dqOegUmKAQAJ and x86 https://www.mail-archive.com/linux-kernel@vger.kernel.org/msg1978069.html)

In the latest patchset I have reversed only the `HAVE_LD_DEAD_CODE_DATA_ELIMINATION` Kconfig instead of the entire original patch which is more manageable.